### ISSUE TITLE
[USABILITY]: Application saved message could be missed by assistive devices

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#182]
## Issue
**UPDATE 3/30/2020** Revising assignees to raise awareness with DEPO leads and VSP design leadership. This is an exploratory issue, but one that could have impact for all forms that can be saved and continued later.

The `Application has been saved` message is appended after the `Back` and `Continue` buttons. This means the message may be missed by assistive device users, who would have a high chance of moving to the next page before having it read aloud. This ticket is not a defect, but only an item to investigate about placing the message before the buttons, or considering appending it to the beginning of next screens, as a way to have it read on first load.

/cc @emilywaggoner for design guidance. This item was flagged as an improvement/enhancement by the 508 office. They're open to seeing what new guidance we come up with and happy to help us re-test.

## Acceptance Criteria

As a screenreader user, I would like to hear the form saved announcement periodically. This could be via `aria-live` attributes, or by positioning it before the buttons and adding a `tabindex="0"` attribute.

## Environment

* Windows 7 64 Bit Enterprise Edition SP1
* Google Chrome
* JAWS 2018.0

---

<img width="578" alt="screen shot 2018-08-22 at 10 58 01 pm" src="https://user-images.githubusercontent.com/934879/44502443-82b7fd00-a65f-11e8-86f1-093cb564acbf.png">

**Comment by 1Copenut** [comment]
Adding some notes from the June 11, 2019 meeting with the VA 508 office. They're working on a character count for a text field, and having similar conversations.

> When a form field edit box has a maximum number of characters allowed it will be considered compliant if one of the following occurs.
>
> 1. The maximum number of characters is indicated in the label.
> 2. There is an audio indication when a user tries to type in more than the max allowed characters and the characters do not appear on the screen.
> 3.  A character count allowed countdown is provided.

The discussion also yielded a couple of possibilities not included above:

1. Could we front-load user expectations by adding a sentence to the first form page stating we'll save user progress on every "Continue"
2. Possibly moving the save message into the progress bar header, and have it appear on the next page after users pressed "Continue"

**Comment by caw310** [comment]
Issue was discussed during 4/2 DSC meeting. Follow up on this is that this issue will be put in the DST backlog for the next sprint (50).  But b/c we (DST) do not have  a11y specialist. Trevor and Ryan or Josh and someone will tackle this to make a proposal.

**Comment by AngelaFowler82** [comment]
Update, during a Slack conversation on 06/04 it was suggested that we put a static message above the form stating that progress is saved every time a change is made. This message will be tagged with a heading so it can easily be skipped if the user does not want to hear it.

**Comment by 1Copenut** [comment]
Quick follow on to Angela's comment, this heading should be above the form and on every page so users can skip it if they choose:

> If it was tagged with the heading and placed above the form, we can put it on every page and the user can simply skip over  it if they don’t want to hear it.

**Comment by AngelaFowler82** [comment]
Thank you for that clarification.

**Comment by 1Copenut** [comment]
I took a quick pass at mocking a message into a logged in form. This would only appear if users are logged in, but would be an SPAN inside the H2 page title.

cc @rtwell 
---

![Screen Shot 2021-06-25 at 4 17 31 PM](https://user-images.githubusercontent.com/934879/123487822-4c034d80-d5d4-11eb-9adc-55388f84cdeb.png)

**Comment by rtwell** [comment]
@1Copenut is this only for non-sighted folks, or for all users?

**Comment by 1Copenut** [comment]
@rtwell All users. By moving it into the H2 at the top of the page, it's read back to screen reader users navigating by headings, (and can be ignored easily) and is also visually present.
